vtm/src/org/oscim/layers/tile/ZoomLimiter.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2018 Gustl22
+ *
+ * This program is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ * PARTICULAR PURPOSE. See the GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.oscim.layers.tile;
+
+public class ZoomLimiter {
+
+    private int mMaxZoom;
+    private int mMinZoom;
+
+    /**
+     * Indicates that layer isn't processed again over zoom limit
+     */
+    private int mZoomLimit;
+
+    private TileManager mTileManager;
+
+    /**
+     * A layer which avoid rendering tiles over a specific zoom limit
+     */
+    public ZoomLimiter(TileManager tileManager, int minZoom, int maxZoom, int zoomLimit) {
+        if (zoomLimit < minZoom || zoomLimit > maxZoom) {
+            throw new IllegalArgumentException("Zoom limit is out of range");
+        }
+        mTileManager = tileManager;
+        mZoomLimit = zoomLimit;
+        mMinZoom = minZoom;
+        mMaxZoom = maxZoom;
+    }
+
+    public void addZoomLimit() {
+        if (mZoomLimit < mMaxZoom && mZoomLimit >= mMinZoom) {
+            // Request tiles of mZoomLimit.
+            mTileManager.addZoomLimit(mZoomLimit);
+        }
+    }
+
+    public void removeZoomLimit() {
+        if (mZoomLimit < mMaxZoom && mZoomLimit >= mMinZoom) {
+            mTileManager.removeZoomLimit(mZoomLimit);
+        }
+    }
```

Comment:
Why having a `ZoomLimiter` with zooms in creation that may be not suitable later in use?
These are `final` variables, so better check them when they set, or there are multiple cases to think?

Reply:
The special case `mZoomLimit == mMaxZoom` in `ZoomLimiter` is the same as the ZoomLimiter wouldn't exist (like current master). So need to include this case in constructor as `ZoomLimiter` replaces all max zoom checks. But this also means that don't need to add extra zoom limit. And if a layer's mMaxZoom isn't as large as TileManager.mMaxZoom, limit wouldn't be [excluded](https://github.com/mapsforge/vtm/commit/f49f0838ad42e803fd54e1fe01cf50906007060a#diff-f5fcd1c424b0ab01b67864b595d23dfcR753).
So need to check if mZoomLimit < mMaxZoom, otherwise there's no need to add zoom limit to `TileManager`

Reply:
Ah ok, thanks for the explanation!
You can PR the change in [issue_528](https://github.com/mapsforge/vtm/tree/issue_528) branch or I can push it later.

Reply:
If you want, you can push it of course :)